### PR TITLE
Add menu button on SplitViewSeparator

### PR DESCRIPTION
**Pull request by sangwoo108**
This menu button shows up a bubble that contains operations for the split view

Resolves https://github.com/brave/brave-browser/issues/38584

image

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Comment by simonhong** [comment]
Please rebase this on to latest master. It's not buildable with it

File: browser/ui/views/split_view/split_view_separator.cc [hunk 1]
```diff
@@ -25,13 +109,52 @@ bool SplitViewSeparator::OnMousePressed(const ui::MouseEvent& event) {
...
+}
+
 void SplitViewSeparator::OnResize(int resize_amount, bool done_resizing) {
   // When mouse goes toward web contents area, the cursor could have been
   // changed to the normal cursor. Reset it resize cursor.
   GetWidget()->SetCursor(ui::Cursor(ui::mojom::CursorType::kEastWestResize));
   if (resize_area_delegate_) {
     resize_area_delegate_->OnResize(resize_amount, done_resizing);
   }
+  LayoutMenuButton();
```
Comment (simonhong) [inline browser/ui/views/split_view/split_view_separator.cc:123]: While resizing, I could see resize bar and menu buttton(bubble) are not synchronously moved.
How about hiding menu button while resizing? or could move both synchronously?

**Review by simonhong — APPROVED** [review]
++ with suggestions 👍🏼

File: browser/ui/views/split_view/split_view_menu_bubble.cc [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+SplitViewMenuBubble* g_bubble = nullptr;
+
+class ItemView : public views::LabelButton {
+  METADATA_HEADER(ItemView, LabelButton)
+ public:
+  ItemView() {
+    auto* ink_drop = views::InkDrop::Get(this);
+    ink_drop->SetMode(views::InkDropHost::InkDropMode::ON);
+    ink_drop->SetBaseColorCallback(base::BindRepeating(
```
[the remark this answers is not in the document]
Comment (sangwoo108) [reply browser/ui/views/split_view/split_view_menu_bubble.cc]: Good call. I cargo culted this from ImageButton but `SetBaseColorId` looks more desirable.

File: browser/ui/views/split_view/split_view_separator.cc (hunk 1 above)
Comment (sangwoo108) [reply browser/ui/views/split_view/split_view_separator.cc]: > How about hiding menu button while resizing?

That's a good idea!

**Review by mkarolin — APPROVED** [review]
`strings` LGTM

**Pull request merged**